pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -141,14 +142,31 @@ func buildAnnotationTextAndData(rule *ngmodels.AlertRule, currentState *state.St
 		for _, k := range keys {
 			values = append(values, fmt.Sprintf("%s=%f", k, currentState.Values[k]))
 		}
-		jsonData.Set("values", simplejson.NewFromAny(currentState.Values))
+		jsonData.Set("values", jsonifyValues(currentState.Values))
 		value = strings.Join(values, ", ")
 	}
 
 	labels := removePrivateLabels(currentState.Labels)
 	return fmt.Sprintf("%s {%s} - %s", rule.Title, labels.String(), value), jsonData
 }
 
+func jsonifyValues(vs map[string]float64) *simplejson.Json {
+	if vs == nil {
+		return nil
+	}
+
+	j := simplejson.New()
+	for k, v := range vs {
+		switch {
+		case math.IsInf(v, 0), math.IsNaN(v):
+			j.Set(k, fmt.Sprintf("%f", v))
```

Comment:
Does this mean the values can now be a combination of strings and floats?

Reply:
Yes, only inside the annotation, not inside anything important like state.
Annotations support arbitrary types here (it just gets rendered wholesale in the UI). I felt this was a much better UX than just dropping the unrepresentable key.

Reply:
What about always rendering the number as a string instead of switching on the value?

Reply:
The UI is looking for a specific thing inside the annotation, unfortunately. I didn't want to fall back all the time, only when necessary, as the annotation seems to prefer numbers. It displays a little nicer, no quotes, etc and IMO is more what users expect.